src/AutoUI/Filters/Filters.tsx

```diff
@@ -96,6 +106,159 @@ const removeFieldsWithNoFilter = (schema: JSONSchema): JSONSchema => {
 	return schema;
 };
 
+/**
+ * Constructs properties for schemas based on their x-ref-scheme or x-foreign-key-scheme annotations.
+ */
+const constructSchemaProperties = (
+	schemaOrProperties: JSONSchemaDefinition,
+	transformedXRefScheme: string[] | undefined,
+	description?: string,
+	title?: string,
+): JSONSchemaDefinition => {
+	if (schemaOrProperties == null || !transformedXRefScheme?.length) {
+		return schemaOrProperties;
+	}
+
+	const [firstRefSchemeKey, ...restRefScheme] = transformedXRefScheme;
+	if (isJSONSchema(schemaOrProperties) && schemaOrProperties.properties) {
+		return {
+			...schemaOrProperties,
+			title: title ?? schemaOrProperties.title,
+			description,
+			properties: constructSchemaProperties(
+				schemaOrProperties.properties,
+				restRefScheme,
+				description,
+			) as JSONSchema['properties'],
+		};
+	}
+	if (isJSONSchema(schemaOrProperties) && schemaOrProperties.items) {
+		return {
+			...schemaOrProperties,
+			title: title ?? schemaOrProperties.title,
+			description,
+			items: constructSchemaProperties(
+				schemaOrProperties.items as JSONSchema,
+				restRefScheme,
+				description,
+			),
+		};
+	}
```

Comment:
nit:
```suggestion
if (isJSONSchema(schemaOrProperties)) {
      if (schemaOrProperties.properties) {
		return {
			...schemaOrProperties,
			title: title ?? schemaOrProperties.title,
			description,
			properties: constructSchemaProperties(
				schemaOrProperties.properties,
				restRefScheme,
				description,
			) as JSONSchema['properties'],
		};
	}
	if (schemaOrProperties.items) {
		return {
			...schemaOrProperties,
			title: title ?? schemaOrProperties.title,
			description,
			items: constructSchemaProperties(
				schemaOrProperties.items as JSONSchema,
				restRefScheme,
				description,
			),
		};
	}
}
```

Reply:
🤦 don't know why I've repeated it... I just had to move it on top, changed it to so that it is not nested